lib/bandit/http1/adapter.ex

```diff
@@ -289,7 +289,7 @@ defmodule Bandit.HTTP1.Adapter do
   defp response_header(nil, status, headers), do: response_header("HTTP/1.0", status, headers)
 
   defp response_header(version, status, headers) do
-    date_header = "date: #{Bandit.HTTP.date_header()}\r\n"
+    headers = [Bandit.Clock.date_header() | headers]
```

Comment:
I wonder if we should be using `Keyword.put_lazy` here in case (for whatever reason) the called Plug has set its own Date header

Reply:
`Keyword` wouldn't work here, since the _keys_ are strings.
We could use `if` + `List.keymember?/2`?

Reply:
Sure!

Reply:
@mtrudel 
The check was a bit more complicated than I thought, since the user defined headers can be any case (e.g. `DaTe`) so I introduced a `has_date_header?/1` function, which uses `Enum.any?/2` with `String.downcase/1`.